app/main.py

```diff
@@ -2,24 +2,74 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, obj: object, objtype: type[object] = None) -> str:
+        return getattr(obj, self.protected_name)
+
+    def __set__(self, obj: object, value: int) -> None:
+        self.validate(value)
+        setattr(obj, self.protected_name, value)
+
+    def validate(self, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Expected value to be an int")
+
+        if value < self.min_amount or value > self.max_amount:
+            raise ValueError(
+                f"Expected {value!r} to be no more than {self.min_amount}"
+                f"and less than {self.max_amount}."
+            )
 
 
 class Visitor:
-   pass
+    def __init__(self, name: int, age: int, height: int, weight: int) -> None:
```

Comment:
In case there are multiple arguments, you could place each on a new line to improve the readability